main.c

```diff
@@ -612,6 +638,15 @@ static void parse_args(int argc, char *const argv[], struct options *options)
     if (optarg_outbound_addr6 != NULL) {
         options->outbound_addr6 = strdup(optarg_outbound_addr6);
     }
+    if (optarg_macaddress != NULL) {
+        if (!options->do_config_network) {
+            fprintf(stderr, "--macaddr cannot be specified when --configure or "
+                            "-c is not specified\n");
+            goto error;
+        } else {
+            options->macaddress = strdup(optarg_macaddress);
+        }
+    }
```

Comment:
@AkihiroSuda Is this what you had in mind?

Reply:
Yes, thanks